### ISSUE TITLE
HW2 Compatibility Issue 

### ISSUE THREAD

**Issue by pkbroot** [danielkraak/GC-Loader#34]
Just got and installed my GC Loader HW2 today and I can't update. It seems to think that firmware "2.0.0.GCLOADER_HW2" is higher than "2.0.1.BETA" perhaps? When I launch the update it just shows the two top lines that display the update and the current detected firmware, but does not give the prompt of press A to start update process. However I still have a notification that a new firmware is present every time I boot up and launch into Swiss. Any suggestions?

_Originally posted by @pkbroot in https://github.com/danielkraak/GC-Loader/issues/32#issuecomment-1668687327_

**Comment by kurohouou** [comment]
So Dan told me the 2.0.1 firmware isn't meant for the new GCLoaders.. maybe 2.0.2 will come out and that will be cross HW supported.. or maybe it will have a whole different chain.  Will have to wait and see.

That said, does your reboot/restart button work?  If I press it, it will just go to a black screen if I was playing a game.. maybe I have something setup wrong?  The soft restart with the Z+A+Start works great though.

Last I am waiting on a new 1TB Micro SD Card to come, the Sandisk I got will not be recognized and won't boot swiss at all (https://a.co/d/glZL7Mx).  Will hopefully find out tomorrow if the older/cheaper sandisk 1tb is better.

**Comment by pkbroot** [comment]
Reset button black screens on mine as well. I also have a small issue where I can't get soft reset to work with swiss set to igr.dol, but if i set it to reboot in swiss that does work.

Also it's rather annoying if 2.0.1 isn't for HW2 but I still get the persistent 'update available' message every time I boot swiss.

**Comment by kurohouou** [comment]
I just have it set to reboot as well.  Would setting it to igr.dol do something else besides just a reboot.  Figure as long as one works i am happy :)

I think Swiss needs to be updated to reflect the change in HW for GCLoader.  Maybe in the coming versions Swiss will know better.

**Comment by pkbroot** [comment]
igr.dol just allows you to reboot to whatever .dol you set on the SD card. I currently have no need to boot to anything other than Swiss so 'reboot' is fine, but it's just a bit of an annoyance knowing it doesn't work.

**Comment by pkbroot** [comment]
If 2.0.1 isnt for use with HW2 it would be really nice if they communicated that somewhere publicly. Dan's shop listing for HW2 links to it for apparently no reason considering the files it links to aren't compatible! GC Loader github has basically no information other than the files,

**Comment by CBal2424** [comment]
You can go into the Swiss ini file I believe and change your GCLoader version to 2.0.1 so the message quits popping up. The only issue I ran into yesterday with my HW2 was that I filled my 256gb memory card and Swiss is still showing 256gb free. This might be more of a Swiss issue. If you want igr.dol to work I suggest saving configurations to a sd2sp2 memory card with igr.dol on there and it will soft reboot just fine. Otherwise HW2 is working great for me.

**Comment by pkbroot** [comment]
Yeah I know, I just haven't done that yet. The message itself is just a minor annoyance. I feel like the bigger issue is the fact that it tells you to update and links to a firmware that is not compatible. That could be a big problem for users. Hopefully they notice and rectify this soon. All these HW2 GC Loaders are likely in the customers hands at this point, and they should probably get in front of this problem asap.

**Comment by tjanas** [comment]
> lIt seems to think that firmware "2.0.0.GCLOADER_HW2" is higher than "2.0.1.BETA"
> 
> _Originally posted by @pkbroot in [#32 (comment)](https://github.com/danielkraak/GC-Loader/issues/32#issuecomment-1668687327)_

Use latest Swiss to get rid of this false firmware update prompt for HW2
https://github.com/emukidid/swiss-gc/releases/tag/v0.6r1527

**Comment by pkbroot** [comment]
Do you guys have any idea yet whether or not HW2 is going to be completely separate with all future firmware updates?

**Comment by pkbroot** [comment]
New version of swiss did indeed fix the update notification issue. However igr.dol soft reset, as well as the physical reset button both still just black screen.

**Comment by tjanas** [comment]
> However igr.dol soft reset… black screen.

Why using igr.dol instead of the Reboot option in Swiss?

**Comment by pkbroot** [comment]
No particular reason, just thought if it's there it should work shouldn't it? But whatever is making it not work seems likely the same reason the reset button doesn't work. Is that a normal thing for GC Loader?

Also is it normal for the gamecube logo splash to only play for like half a second and then skip right to Swiss boot? I was using whatever was the most recent version (as of yesterday) and the whole logo animation played out before booting into Swiss. When I updated today, it began skipping right to Swiss after the animation played just the very beginning.

**Comment by kurohouou** [comment]
If you use the files in the ISO folder instead of the GCLoader folder you will get the full boot logo sequence I found.  In my case I took swiss_r1527(ntsc-u).iso and renamed to boot.iso and put it in the root of the sdcard and it all worked fine then.  Not quite sure exactly what the difference is between the two, they both seem to work just as well, its just the ones in the ISO folder have the full boot logo sequence.

**Comment by pkbroot** [comment]
I just found this answer in another old post on reddit. Thanks for helping though. I also think i realized that the igr.dol reset setting is probably for people using some exploited save or similar method to launch swiss, so they don't have to do the exploit again each time they reset. However I still would like to know if there is any solution to getting the physical reset button to work. It's not a major problem but it would be nice if it was possible to retain functionality.

**Comment by tjanas** [comment]
> Also is it normal for the gamecube logo splash to only play for like half a second and then skip right to Swiss boot?

Yes, that is normal and has been part of [Swiss](https://github.com/emukidid/swiss-gc) for quite awhile now.

Reset is more reliable than using igr.dol. If you want to see the full GameCube boot sequence you could do that when you launch a game from Swiss based on your Swiss settings (I think it is “Enable boot through IPL” in advanced settings).

**Comment by pkbroot** [comment]
I am good with not seeing the boot sequence again. I was just hoping the reset button would work but it just black screens and I have to power cycle the system.

**Comment by tjanas** [comment]
When you are in-game, what happens if you use the Swiss Reset controller hotkey (R+Z+B+dpadDown) and have Swiss configured to Reset?

**Comment by kurohouou** [comment]
When I have Swiss In-Game reset set to Reboot, the Z+A+Start button reboots my GC.  But the physical reboot button goes to a black screen.. think that's the only major issue I see right now with the new GCLoader.

**Comment by pkbroot** [comment]
Yeah in game reset works fine if Swiss is set to reboot. And the last post is correct, the button combo is Z+A+Start now.

It's the reset button on the console that doesn't work.

**Comment by tjanas** [comment]
> the button combo is Z+A+Start now.
> 

Both combinations (Z+A+Start and R+Z+B+dpadDown) are valid:
```
#define PAD_COMBO_EXIT1   (PAD_BUTTON_B | PAD_BUTTON_R | PAD_BUTTON_Z | PAD_BUTTON_DOWN)
#define PAD_COMBO_EXIT2   (PAD_BUTTON_A | PAD_BUTTON_Z | PAD_BUTTON_START)
#define PAD_COMBO_ORIGIN  (PAD_BUTTON_Y | PAD_BUTTON_X | PAD_BUTTON_START)
#define PAD_COMBO_RESET   (PAD_BUTTON_B | PAD_BUTTON_X | PAD_BUTTON_START)
#define PAD_COMBO_RESTART (PAD_BUTTON_R | PAD_BUTTON_Z | PAD_BUTTON_START)
```
https://github.com/emukidid/swiss-gc/blob/master/cube/patches/base/dolphin/pad.h

**Comment by pkbroot** [comment]
OK but thats not helpful at all in relation to the problem of the reset button not working. Already cleared up in game reset issues several posts back.

**Comment by kurohouou** [comment]
Well this isn't good.  I have tried 2 different Sandisk 1TB cards, neither are recognizable by the new GCLoader :(  Hopefully this can be fixed soon!

**Comment by danielkraak** [comment]
Do you perhaps have exact names and brands for the cards you tried? An amazon link could also be helpful.

I am going to have to test the card myself.

Edit: also feel free to contact me through the contact form so we can further diagnose the issue on your end. https://gc-loader.com/contact-us/

**Comment by kurohouou** [comment]
Yep, these are the two I tried,

https://a.co/d/deghaW5

and 

https://a.co/d/7v1lwkc

Looking forward to hearing your results!

**Comment by danielkraak** [comment]
Could you contact me through the contact form so I can do some suggestions to diagnose the issue, as I have to await the order of the 1tb card.

**Comment by kurohouou** [comment]
Message sent

**Comment by danielkraak** [comment]
Did you use this form? https://gc-loader.com/contact-us/

I did not receive anything yet. If you did, perhaps it takes a while.

**Comment by kurohouou** [comment]
Oh sorry I submitted it through the other site.  You should see it now.

**Comment by icd2k3** [comment]
I have a SanDisk Extreme 512gb and TeamGroup GO 256gb that both don't work in the new GCLoader

**Comment by danielkraak** [comment]
I ordered a sandisk that should arrive tomorrow.

In the meantime, could you check if the microcontroller boots correctly on the gc loader. You can tell as follows:

You can tell if the microcontroller boots, by looking at the LED. When you press power, the LED should turn on, but it will be dim (this is before any blinking occurs, which indicates the gc and gc loader are transferring data). When the microcontroller manages to boot, it should go brighter. You can check this with an SD Card that works or by not inserting an SD Card at all (in this case, the GameCube should report that no disc has been inserted in the main menu). If the LED stays dim with the not working SD Card, then this may be a hardware issue.

**Comment by icd2k3** [comment]
thanks @danielkraak - here's a video of the SanDisk Extreme 512gb with it inserted AND not inserted

https://www.dropbox.com/scl/fi/h6d67t7oq5oqhhiwzrx8r/IMG_0805.mov?rlkey=2xq8mqyocnde34si4oyng74rg&dl=0

while inserted, the light is dim (no blinking, bright, or changes in status). When it is **not** inserted it starts dim, but then goes brighter... so is my hardware ok?

TeamGroup Go 256gb and Lexar Play 1TB both are NOT recognized... instead of black screen it just goes to GC bios as if no card was inserted

**Comment by kurohouou** [comment]
Confirmed with my 1TB MicroSD Card and 512GB MicroSD Card.  

The 1TB has a dim light when powered on with NO boot into Swiss

The 512GB has a bright light and boots into Swiss fine

**Comment by danielkraak** [comment]
Thanks for the information. I was able to replicate the issues on my end. There will be some more information hopefully soon.

**Comment by tjanas** [comment]
HW2 firmware update: https://gc-loader.com/firmware-updates-hw2/
